### ISSUE TITLE
Crash when trying to share folder

### ISSUE THREAD

**Issue by jay-tuckey** [nextcloud/android#8166]
### Steps to reproduce
1. Open Nextcloud App
2. Tap the Share icon on folder
3. Type the name of the other user (on same server)

### Expected behaviour
It should share to the other use

### Actual behaviour
I get an immediate crash. If I re-try it crashes again.
Sharing from the web interface works fine.

### Can you reproduce this problem on https://try.nextcloud.com?
Yes, just tried on try.nextcloud.com and got the same crash sharing to a random account there.

### Environment data
Android version: 11

Device model: S21 Ultra

Stock or customized system: Stock

Nextcloud app version: 30150090 (installed from F-Droid)

Nextcloud server version: 20.0.8 - running in docker using the official nextcloud docker

Reverse proxy: Yes - nginx

#### Web server error log
```
root@bigtuckey #~> cat nextcloud_crash_log
Attaching to nextcloud_nextcloud_1
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:09:08 +0000] "GET /index.php/204 HTTP/1.0" 204 1347 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:09:08 +0000] "GET /index.php/204 HTTP/1.0" 204 1343 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:09:08 +0000] "GET /index.php/204 HTTP/1.0" 204 1347 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:09:08 +0000] "GET /index.php/204 HTTP/1.0" 204 1347 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:13 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:25 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:25 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:29 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:09:31 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.0" 304 249 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:43 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v1.php/cloud/user?format=json HTTP/1.0" 200 1330 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /status.php HTTP/1.0" 200 1068 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v2.php/cloud/capabilities?format=json HTTP/1.0" 304 251 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /status.php HTTP/1.0" 200 1068 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v2.php/cloud/capabilities?format=json HTTP/1.0" 304 251 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v2.php/apps/user_status/api/v1/predefined_statuses?format=json HTTP/1.0" 200 1789 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v2.php/apps/user_status/api/v1/predefined_statuses?format=json HTTP/1.0" 200 1785 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v1.php/cloud/user?format=json HTTP/1.0" 200 1330 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:46 +0000] "GET /ocs/v1.php/cloud/user?format=json HTTP/1.0" 200 1330 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "PROPFIND /remote.php/webdav/ HTTP/1.0" 207 1444 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "PROPFIND /remote.php/webdav/ HTTP/1.0" 207 1444 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "PROPFIND /remote.php/webdav/ HTTP/1.0" 207 32090 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "PROPFIND /remote.php/webdav/ HTTP/1.0" 207 32090 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "GET /ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.0" 200 4371 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:47 +0000] "GET /ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.0" 200 4371 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:50 +0000] "GET /ocs/v2.php/apps/activity/api/v2/activity/filter?object_type=files&format=json&previews=true&sort=desc&object_id=457387 HTTP/1.0" 200 2067 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:50 +0000] "PROPFIND /remote.php/dav/versions/jtuckey/versions/457387 HTTP/1.0" 404 912 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:51 +0000] "GET /ocs/v2.php/apps/activity/api/v2/activity/filter?object_type=files&format=json&previews=true&sort=desc&object_id=457387&since=70360 HTTP/1.0" 304 1007 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:51 +0000] "PROPFIND /remote.php/dav/versions/jtuckey/versions/457387 HTTP/1.0" 404 912 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:55 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:52 +0000] "GET /ocs/v2.php/apps/files_sharing/api/v1/sharees?format=json&itemType=file&search=s&page=1&perPage=50 HTTP/1.0" 200 4514 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:57 +0000] "POST /ocs/v2.php/apps/files_sharing/api/v1/shares HTTP/1.0" 403 927 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:57 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:09:55 +0000] "GET /ocs/v2.php/apps/files_sharing/api/v1/sharees?format=json&itemType=file&search=samara&page=1&perPage=50 HTTP/1.0" 200 1660 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.15.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:01 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:10:01 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.0" 304 249 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:13 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:25 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:29 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:10:31 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.0" 304 249 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:33 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:43 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:10:55 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 995 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:11:01 +0000] "PUT /apps/user_status/heartbeat HTTP/1.0" 204 686 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:11:01 +0000] "GET /ocs/v2.php/apps/user_status/api/v1/user_status HTTP/1.0" 200 922 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:11:01 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - - [12/Mar/2021:23:11:02 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.0" 304 249 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0"
nextcloud_1_38cb3d836e7c | 10.1.1.9 - jtuckey [12/Mar/2021:23:11:05 +0000] "PROPFIND /remote.php/dav/files/jtuckey/ HTTP/1.0" 207 1008 "-" "Mozilla/5.0 (Linux) mirall/3.1.3git (build 4849) (Nextcloud)"
```

#### Nextcloud log (data/nextcloud.log)
```
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Accessing result data after operation failed!
	at com.owncloud.android.lib.common.operations.RemoteOperationResult.getData(RemoteOperationResult.java:486)
	at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForCreateShareOperations(ErrorMessageAdapter.java:254)
	at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForResultAndOperation(ErrorMessageAdapter.java:143)
	at com.owncloud.android.utils.ErrorMessageAdapter.getErrorCauseMessage(ErrorMessageAdapter.java:79)
	at com.owncloud.android.ui.activity.FileActivity.onRemoteOperationFinish(FileActivity.java:380)
	at com.owncloud.android.ui.activity.FileDisplayActivity.onRemoteOperationFinish(FileDisplayActivity.java:1704)
	at com.owncloud.android.services.OperationsService.lambda$dispatchResultToOperationListeners$0(OperationsService.java:734)
	at com.owncloud.android.services.-$$Lambda$OperationsService$KsB1LbjHjlEBSp0h7XO24czs-OQ.run(Unknown Source:6)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8506)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30150090
Build flavor: generic

************ DEVICE INFORMATION ************
Brand: samsung
Device: z3s
Model: SM-G988B
Id: RP1A.200720.012
Product: z3sxxx

************ FIRMWARE ************
SDK: 30
Release: 11
Incremental: G988BXXU6DUB5
```

**Comment by AndyScherzinger** [comment]
Fixed via the latest stable 3.15.1 which f-droid should publish with the next days (hopefully)

**Comment by jay-tuckey** [comment]
Yup, @AndyScherzinger can confirm its fixed.
